initrd/bin/seal-hotpkey

```diff
@@ -73,20 +80,29 @@ else
 	HOTPKEY_BRANDING="HOTP USB Security Dongle"
 fi
 
-echo -e ""
-read -s -p "Enter your $HOTPKEY_BRANDING Admin PIN: " admin_pin
-echo -e "\n"
-
-hotp_initialize "$admin_pin" $HOTP_SECRET $counter_value "$HOTPKEY_BRANDING"
+# try using factory default admin PIN
+admin_pin="12345678"
+hotp_initialize "$admin_pin" $HOTP_SECRET $counter_value "$HOTPKEY_BRANDING" >/dev/null 2>&1
```

Comment:
Still unconformable with this one.
Could we use `gpg --card-status | grep "Signature counter"` and only have that code run when its under 5 or something (when the USB security dongle was just received?) From my experience, 3 ADMIN PIN attempts is not a lot in the absence of having a Reset code (Reset code can unlock Admin PIN which can unlock User PIN) but as of today, without a private key backup, I had a lot of reports of users locking themselves out. Running this at each HOTP reseal is augmenting the chances a user locks himself out of his USB dongle. 

I would be ok running that code and detecting default Admin PIN being used if the counter signature is low, only.

Reply:
I like the idea of conditionally trying this.  The signature counter idea might work, I think the key there is to smooth out the UX.  If the user neglects to change the PIN (obviously a bad thing), then it'd be even worse if much later they are suddenly asked for a PIN they don't know.  5 signatures later could be a long time after the OEM reset occurred and told them the default PIN.

Maybe we could phase it?
* 0-4 signatures - try default, maybe suggest a bit more strongly to change it (and maybe suggest how since there is no UI for this currently)
* 5-9 signatures - don't try default, but remind user what it was (so at least they get used to entering the default PIN)
* 10+ signatures - don't try default and don't remind, to minimize nuisance reminders in the long run

I wonder if we could also get the number of attempts remaining before making an attempt, to be sure this doesn't consume the last attempt if you had mistyped the pin once or twice and aborted.

Reply:
@JonathonHall-Purism 
> I wonder if we could also get the number of attempts remaining before making an attempt, to be sure this doesn't consume the last attempt if you had mistyped the pin once or twice and aborted.

Yes we can! This is `PIN retry counter : 3 3 3`

I love the phasing idea. I would not have any objection into having that upstream, but main concern is really to consume even 1 Admin attempt if the opengpg card was used before to sign other stuff: meaning the user might use the smartcard to encrypt+sign, where we can take assumption that if counter > 5 is definitely used...


Throwing additional idea here: maybe time is a better factor. If the keypair was enerated recently by oem, the key "created" date being recent (check Purism numbers, but is there a delivery that happens after 1 month?)  with a signing counter <=3 should mean we can consume and Admin PIN attempt.

My point here is that it is better sooner then later to have the user reown hist gpg card then later. In the user case a user is actually using his USB Security dongle for something else then detach signing /boot and HOTP, we can think of that user as being one that would run the reownership wizard anyway?

Reply:
Got this working, it only tries the default PIN for up to 1 month and only if >= 3 attempts remain (my keys default to 3).  If it doesn't try the default for either reason, it shows a reminder.

I couldn't use `gpg --card-status` to do this as it locked up the keys (https://github.com/Nitrokey/nitrokey-pro-firmware/issues/54) but I was able to do it by getting the key creation time from Heads' copy of the public key, and getting the PIN counter from `hotp_verification info`.